utils/proxy.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import os
+from urllib import getproxies
+from urlparse import urlparse
+
+log = logging.getLogger(__name__)
+
+def set_no_proxy_settings():
+    # Starting with Agent 5.0.0, there should always be a local forwarder
+    # running and all payloads should go through it. So we should make sure
+    # that we pass the no_proxy environment variable that will be used by requests
+    # See: https://github.com/kennethreitz/requests/pull/945
+    to_add = ["127.0.0.1", "localhost", "169.254.169.254"]
+    no_proxy = os.environ.get("no_proxy", "")
+    if not no_proxy.strip():
+        no_proxy = []
+    else:
+        no_proxy = no_proxy.split(',')
+
+    for host in to_add:
+        if host not in no_proxy:
+            no_proxy.append(host)
+
+    os.environ['no_proxy'] = ','.join(no_proxy)
+
+def get_proxy(agentConfig):
+    proxy_settings = {}
+
+    # First we read the proxy configuration from datadog.conf
+    proxy_host = agentConfig.get('proxy_host', None)
+    if proxy_host is not None:
+        proxy_settings['host'] = proxy_host
+        try:
+            proxy_settings['port'] = int(agentConfig.get('proxy_port', 3128))
+        except ValueError:
+            log.error('Proxy port must be an Integer. Defaulting it to 3128')
+            proxy_settings['port'] = 3128
+
+        proxy_settings['user'] = agentConfig.get('proxy_user', None)
+        proxy_settings['password'] = agentConfig.get('proxy_password', None)
+        log.debug("Proxy Settings: %s:%s@%s:%s" % (proxy_settings['user'],
+            "*****", proxy_settings['host'], proxy_settings['port']))
```

Comment:
nitpick: The "standard" way is to either pass the interpolation variables as the variadic args in log or use a format.
```python
log.debug("%s %s", string1, string2)
# or
log.debug("{0} {1}".format(string1, string2))
```